snapcraft/parts/plugins/kernel_plugin.py

```diff
@@ -193,35 +192,35 @@
 }
 
 
-class KernelPluginProperties(plugins.PluginProperties, plugins.PluginModel):
+class KernelPluginProperties(plugins.PluginProperties, frozen=True):
     """The part properties used by the Kernel plugin."""
 
-    kernel_kdefconfig: List[str] = ["defconfig"]
-    kernel_kconfigfile: Optional[str]
-    kernel_kconfigflavour: Optional[str]
-    kernel_kconfigs: Optional[List[str]]
-    kernel_image_target: Any
+    kernel_kdefconfig: list[str] = ["defconfig"]
+    kernel_kconfigfile: str | None = None
+    kernel_kconfigflavour: str | None = None
+    kernel_kconfigs: list[str] | None = None
+    kernel_image_target: Any = None
```

Comment:
Based on the validator below:

```suggestion
    kernel_image_target: str | dict[str, Any] | None = None
```